app/components/ui/CalculatorInput.tsx

```diff
@@ -64,10 +65,31 @@ const CalculatorInput = () => {
       body: JSON.stringify(data),
     });
     const processedData = await response.json();
+    if (!response.ok) return handleErrors(processedData.error);
+
     setData(processedData);
     setView("dashboard");
   };
 
+  const handleErrors = (error: APIError) => {
+    switch (error.code) {
+      case "ITL3_NOT_FOUND":
+      case "INSUFFICIENT_PRICES_PAID_DATA":
+        methods.setError(
+          "housePostcode", 
+          { message: 
+            "Insufficient data for this postcode. Please try again with a different postcode" 
+          }
+        );
+        break;
```

Comment:
We could handle these two cases differently, but right now this is a little white-lie about non-English postcodes.